File: Commons/Action/src/main/java/com/greentree/commons/action/observer/integer/IntAction.java
package com.greentree.commons.action.observer.integer;

import java.util.function.IntConsumer;

import com.greentree.commons.action.MultiAction;
import com.greentree.commons.action.observable.IntObservable;
import com.greentree.commons.action.observer.IntObserver;

public class IntAction extends MultiAction<IntConsumer> implements IntObserver, IntObservable {
	private static final long serialVersionUID = 1L;

	@Override
	public void event(int i) {
		for(var l : listeners)
			l.accept(i);
	}

}

File: Commons/Action/src/main/java/com/greentree/commons/action/observer/integer/IntAction.kt
package com.greentree.commons.action.observer.integer

import com.greentree.commons.action.MultiAction
import com.greentree.commons.action.observable.IntObservable
import com.greentree.commons.action.observer.IntObserver
import java.util.function.IntConsumer

class IntAction : MultiAction<IntConsumer>(), IntObserver, IntObservable {

	override val listenerCount: Int
		get() = listeners.size()

	override fun event(i: Int) {
		for(l in listeners)
			l.accept(i)
	}
}
